test/testenv/testenv.go

```diff
@@ -0,0 +1,82 @@
+package testenv
+
+import (
+	"io"
+	"net"
+
+	"google.golang.org/grpc"
+	mixerpb "istio.io/api/mixer/v1"
+	"istio.io/mixer/cmd/server/cmd"
+	"istio.io/mixer/cmd/shared"
+	"istio.io/mixer/pkg/handler"
+	"istio.io/mixer/template"
+)
+
+// TestEnv interface
+type TestEnv interface {
+	io.Closer
+	CreateMixerClient() (mixerpb.MixerClient, *grpc.ClientConn, error)
+}
+
+type testEnv struct {
+	addr         string
+	mixerContext *cmd.ServerContext
+	shutdown     chan struct{}
+}
+
+// Args includes the required args to initialize Mixer server.
+type Args struct {
+	ConfigStoreURL                string
+	ConfigStore2URL               string
+	ConfigDefaultNamespace        string
+	ConfigIdentityAttributeDomain string
+}
+
+// NewTestEnv creates a TestEnv instance.
+func NewTestEnv(args *Args, adapters []handler.InfoFn) (TestEnv, error) {
+	lis, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		return nil, err
+	}
+
+	context := cmd.SetupTestServer(template.SupportedTmplInfo, adapters, args.ConfigStoreURL, args.ConfigStore2URL,
+		args.ConfigDefaultNamespace, args.ConfigIdentityAttributeDomain)
+	shutdown := make(chan struct{})
+
+	go func() {
+		shared.Printf("Start test mixer on:%v", lis.Addr())
```

Comment:
should we print mixer closing as well.. Consider pasting the output of the console (same that adapter developer will see) for the sample test you have.

Reply:
Shutdown message is printed in the if-block below.